examples/complex_struct_deserialization.rs

```diff
@@ -72,3 +75,19 @@ fn main() -> Result<(), EdnError> {
 
     Ok(())
 }
+
+fn main() -> Result<(), EdnError> {
+    complex_ok()?;
+    complex_wrong()?;
+    Ok(())
+}
+
+#[test]
+fn test_complex_ok() {
+    let _ = complex_ok();
```

Comment:
I don't get these new tests on the examples, isn't this duplicate code?

Also, if it isn't, shouldn't this be unwrapping or asserting it is an Ok value? Because the way it is, it seem to be ignoring the result, and the test could never fail

Reply:
The `main` function is for the `cargo run --examples <example_name>` while `test` is for the `cargo test --examples`. We run `cargo test --examples` in the CI while a user can run `cargo run --examples <example_name>` to understand the example

Reply:
The Result of the function call doesn't ignore the assert, it just happens before the return, so if the test fails the example will fail as well.